src/Liquifier.sol

```diff
@@ -176,13 +179,24 @@ contract Liquifier is Initializable, UUPSUpgradeable, OwnableUpgradeable, Pausab
         return eEthShare;
     }
 
+    function setFeeSwappingEETHToSTETH(uint128 _fee) external onlyOwner {
+        require(_fee <= 10**18, "INVALID_FEE");
+        feeSwappingEETHToSTETH = _fee;
+    }
     // Swap user's eETH for Liquifier's stETH
     function swapEEthForStEth(uint256 _amount) external whenNotPaused nonReentrant {
-        if (_amount > lido.balanceOf(address(this))) revert NotEnoughBalance();
-        if (_amount > liquidityPool.eETH().balanceOf(msg.sender)) revert NotEnoughBalance();
-
-        IERC20(address(liquidityPool.eETH())).safeTransferFrom(msg.sender, address(this), _amount);
+        bool isWhitelistedSwapper = roleRegistry.hasRole(EETH_STETH_SWAPPER, msg.sender);
+        uint256 fees = 0;
+        if (!isWhitelistedSwapper) {
+            fees = _amount * feeSwappingEETHToSTETH / 10**18;
+        }
+        _amount -= fees;
+        feeAccumulated += fees;
```

Comment:
Even though we have the nonReentrant guard on this function, could you rewrite it to use the Check-Effects-Interact pattern. It is nice to have for readability and consistency in the codebase  

https://docs.soliditylang.org/en/latest/security-considerations.html